src/istio/control/client_context_base.h

```diff
@@ -17,24 +17,37 @@
 #define ISTIO_CONTROL_CLIENT_CONTEXT_BASE_H
 
 #include "include/istio/mixerclient/client.h"
+#include "include/istio/utils/attribute_names.h"
+#include "include/istio/utils/local_attributes.h"
 #include "mixer/v1/config/client/client_config.pb.h"
 #include "request_context.h"
 
 namespace istio {
 namespace control {
 
+bool isOutbound(const google::protobuf::Map<
```

Comment:
I suggest
1) move this isOutbound function to istio/util so it can be called by http and tcp
2) It takes "const ::istio::mixer::v1::Attributes&",  not its internal attributes map
3) ClientContextBase, take a "bool  is_outbound" in its constructor.   Or add "bool is_outbound" into LocalAttributes class since it is the one using it.

Reply:
done.